lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -1732,6 +1740,26 @@ def set_log_entry_fields(record, entry)
       end
     end
 
+    # Parse labels. Return nil if not set.
+    def parse_labels(record)
+      payload_labels = record.delete(@labels_key)
+      unless payload_labels.is_a?(Hash)
+        @log.error "Invalid value of '#{@labels_key}' in the payload: " \
+                   "#{payload_labels}. Labels need to be a JSON object.", err
+        return nil
+      end
+      payload_labels.each do |k, v|
```

Comment:
[Optional] The following might be somewhat more helpful to users:
```ruby
non_string_keys = payload_labels.each_with_object([]) do |k, v, a|
  a << k unless k.is_a?(String) && v.is_a?(String)
end
unless non_string_keys.empty?
  @log.error "Invalid value of '#{@labels_key}' in the payload: " \
             "#{payload_labels}. Labels need string values for all keys; " \
             "keys #{non_string_keys} don't."
  return nil
end
```

Reply:
I'm leaning towards the current implementation, which exits early and performs minimum calculation for performance concerns. Since we will print `payload_labels`, it should be sufficient for users to figure out what went wrong.